src/database/database-sqlite3.cpp

```diff
@@ -191,40 +216,68 @@ MapDatabaseSQLite3::MapDatabaseSQLite3(const std::string &savedir):
 
 MapDatabaseSQLite3::~MapDatabaseSQLite3()
 {
-	FINALIZE_STATEMENT(m_stmt_read)
-	FINALIZE_STATEMENT(m_stmt_write)
-	FINALIZE_STATEMENT(m_stmt_list)
-	FINALIZE_STATEMENT(m_stmt_delete)
+	FINALIZE_STATEMENT(read)
+	FINALIZE_STATEMENT(write)
+	FINALIZE_STATEMENT(list)
+	FINALIZE_STATEMENT(delete)
 }
 
 
 void MapDatabaseSQLite3::createDatabase()
 {
-	assert(m_database); // Pre-condition
+	assert(m_database);
 
-	SQLOK(sqlite3_exec(m_database,
+	// Note: before 5.12.0 the format was blocks(pos INT, data BLOB).
+	// This function only runs for newly created databases.
+
+	const char *schema =
 		"CREATE TABLE IF NOT EXISTS `blocks` (\n"
-			"	`pos` INT PRIMARY KEY,\n"
-			"	`data` BLOB\n"
-			");\n",
-		NULL, NULL, NULL),
+			"`x` INTEGER,"
+			"`y` INTEGER,"
+			"`z` INTEGER,"
+			"`data` BLOB NOT NULL,"
+			// Declaring a primary key automatically creates an index and the
+			// order largely dictates which range operations can be sped up.
+			// see also: <https://www.sqlite.org/optoverview.html#skipscan>
+			// Putting XZ before Y matches our MapSector abstraction.
+			"PRIMARY KEY (`x`, `z`, `y`)"
```

Comment:
The old index (where keys come from `getBlockAsInteger`) has used a different order (z,y,x). Maybe worth noting down somewhere.
This change is fine I guess.

```cpp
s64 MapDatabase::getBlockAsInteger(const v3s16 &pos)
{
	return (u64) pos.Z * 0x1000000 +
		(u64) pos.Y * 0x1000 +
		(u64) pos.X;
}
```

Reply:
> Maybe worth noting down somewhere.

I disagree.